src/main/java/org/zalando/baigan/service/EtcdConfigurationRepository.java

```diff
@@ -16,22 +19,23 @@
 /**
  * @author mchand
  */
+// TODO Upgrade to v3 and add E2E test
+public class EtcdConfigurationRepository {
 
-public class EtcdConfigurationRepository extends AbstractConfigurationRepository {
-
+    private static final ObjectMapper objectMapper = new ObjectMapper().registerModule(new GuavaModule());
     private static final String ETCD_URL_ENV_NAME = "ETCD_URL";
     private static final String CONFIG_PATH_PREFIX = "/v2/keys/";
     private Logger LOG = LoggerFactory.getLogger(EtcdConfigurationRepository.class);
     private EtcdClient etcdClient;
 
     @VisibleForTesting
-    public EtcdConfigurationRepository(final EtcdClient etcdClient) {
+    public EtcdConfigurationRepository(final EtcdClient etcdClient, final BaiganConfigClasses baiganConfigClasses) {
```

Comment:
To be able to do this, I first need end-2-end tests. This, however, requires upgrading to etcd v3, since I failed to get a docker image running with Testcontainers for version 2. I would propose to first make a separate PR to upgrade etcd to version 3, then proceed to implementing structured config types in this PR. What do you think?

Reply:
I extracted the test coverage improvements to a separate PR https://github.com/zalando-stups/baigan-config/pull/80 to further slim down this one here. When we have a decision on the etcd repo, I will potentially make a separate PR with version upgrade and tests.

Reply:
Sounds good to me - also I am not necessarily saying that we need to block this PR until we have resolved all other possible dependencies, totally fair to me if we say "this and that become TODOs" :next_track_button: